algorithms.py

```diff
@@ -2195,6 +2688,32 @@ def verify(self):
                         self.gc_map,
                         self.get_gc_left_bound,
                     )
+        if self.model == "smc_k":
+            self.verify_hulls()
+
+
+def intersect_hulls(a, b):
```

Comment:
What about adding an ``intersects_with`` method to Hull. so would be ``a.intersects_with(b)``?